src/query/index_task.rs

```diff
@@ -0,0 +1,111 @@
+use crate::cluster::{Cluster};
+use std::sync::Arc;
+use crate::errors::{ErrorKind, Result};
+use crate::{ResultCode};
+
+
+
+
+/// Instantiate an IndexTask instance to query Aerospike database cluster for index creation
+/// status.
+#[derive(Debug, Clone)]
+pub struct IndexTask {
+    cluster: Arc<Cluster>,
+    namespace: String,
+    index_name: String
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum IndexTaskResult {
+    NotFound,
+    InProgress,
+    Complete
+}
```

Comment:
Maybe we should call this the task "status", rather than the task "result". The result of the index task, is that the index is either created successfully or that an error occurred.

Reply:
Sounds good.